packages/components/src/calendar/date-range-calendar/index.tsx

```diff
@@ -151,13 +155,20 @@ export const DateRangeCalendar = ( {
 		mode: 'range',
 	} );
 
+	const onChange: OnSelectHandler< typeof selectedProp > = useCallback(
+		( selected, triggerDate, modifiers, e ) => {
+			// Convert internal `null` to `undefined` for the public event handler.
+			onSelect?.( selected ?? undefined, triggerDate, modifiers, e );
+		},
+		[ onSelect ]
+	);
+
 	const [ selected, setSelected ] = useControlledValue<
 		DateRange | undefined | null
```

Comment:
Nit: could this be also changed to `typeof selectedProp` ?

Reply:
As a further improvement, it would be nice to make the `useControlledValue` hook `infer` the `T` type from the received arguments, but that's definitely something we don't have to work on today.

Reply:
> Nit: could this be also changed to typeof selectedProp ?

Abosolutely, changed it to `typeof selectedProp` in 8c3977fb081b15d9d985a3701f1ab0224c64f853. 🙌